parsing/builtin_attributes.ml

```diff
@@ -92,6 +92,16 @@ let alert_attr x =
       Some (x, "deprecated", string_of_opt_payload x.attr_payload)
   | "ocaml.alert"|"alert" ->
       begin match kind_and_message x.attr_payload with
+      | Some ("polymorphic_comparison" as kind, message) ->
+          let message = match message with
```

Comment:
Is it really worth special casing this new alert here?  I guess this is to avoid duplicating the more verbose message everywhere.  But someone explicitly enabling the alert would know what it means, just seeing its name, no?

Reply:
I personally think it is worth the special case here to avoid duplication.

> But someone explicitly enabling the alert would know what it means, just seeing its name, no?

To me, the rational is that if this is enabled in a medium-size or bigger project with several people and someone is contributing for example, they might not be familiar with what the alert means and might be discouraged by only seeing
```
File "alert_polymorphic_comparison.ml", line 12, characters 9-16:
12 | let _a = compare
              ^^^^^^^
Alert polymorphic_comparison: Stdlib.compare
```
for example, if it there was no explanation.

Reply:
Fair enough.  What about only replacing an empty payload with the full message?  If one later introduce functions parametrized over a comparison function, one might want to provide explicit messages to the "legacy" version, and adding the generic message would feel weird.

Reply:
I am not completely convinced by the argument for the special casing. Copying the alert message is certainly boring when writing the PR, but this is an one time issue. I would rather keep the alert as vanilla as possible. If this ends up creating too much maintenance pain, it is probably a sign that we should improve the interface for all alerts (and not only the `polymorphic_comparison` one).